pandas/tseries/resample.py

```diff
@@ -673,18 +670,15 @@ def aggregate(self, arg, *args, **kwargs):
     def _get_new_index(self):
         """ return our new index """
         ax = self.ax
-        ax_attrs = ax._get_attributes_dict()
-        ax_attrs['freq'] = self.freq
-        obj = self._selected_obj
 
         if len(ax) == 0:
-            new_index = PeriodIndex(data=[], **ax_attrs)
-            return obj.reindex(new_index)
-
-        start = ax[0].asfreq(self.freq, how=self.convention)
-        end = ax[-1].asfreq(self.freq, how='end')
+            values = []
+        else:
+            start = ax[0].asfreq(self.freq, how=self.convention)
+            end = ax[-1].asfreq(self.freq, how='end')
```

Comment:
> ahh, so this should actually be returning a PeriodIndex where freq is 5D, then the convention wouldn't matter as we are not going to a DTI.

:+1: 

> I think we need to be more explicit about returning PeriodIndex then, right?

Yes I would vote to only return `PeriodIndex`, unless there's an explicit conversion. Rules (here)[https://github.com/pydata/pandas/pull/12874#discussion_r59448140]. 

The additional feature is how to deal with bins that aren't strict subperiods or superperiods. e.g. resampling from monthly to weekly. That's a bigger question I think, probably not for this PR